src/platform/Ameba/AmebaOTAImageProcessor.cpp

```diff
@@ -173,13 +176,16 @@ void AmebaOTAImageProcessor::HandleProcessBlock(intptr_t context)
         return;
     }
 
+    ByteSpan block   = imageProcessor->mBlock;
+    CHIP_ERROR error = imageProcessor->ProcessHeader(block);
+
 #if defined(CONFIG_PLATFORM_8721D)
     if (!imageProcessor->readHeader) // First block received, process header
     {
         uint8_t * tempBuf          = (uint8_t *) ota_update_malloc(32);
         imageProcessor->pOtaTgtHdr = (update_ota_target_hdr *) ota_update_malloc(sizeof(update_ota_target_hdr));
 
-        memcpy(tempBuf, imageProcessor->mBlock.data(), 32);
+        memcpy(tempBuf, block.data(), 32);
```

Comment:
How do you know block.size() >= 32 after processing the Matter header?

Reply:
Thanks for pointing this out. This has been fixed.